<?xml version="1.0"?>
<robot name="Bombel1" xmlns:xacro="http://ros.org/wiki/xacro">

<xacro:property name="a2_length" value="$(arg a2_length)" />
<xacro:property name="a3_length" value="$(arg a3_length)" />

 <material name="blue">
   <color rgba="0 0 0.8 1"/>
 </material>

 <material name="yellow">
   <color rgba="255 255 0 1"/>
 </material>

 <material name="green">
   <color rgba="0 255 0 1"/>
 </material>


  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.3"/>
      </geometry>
      <material name="blue"/>
      <origin xyz="0 0 0.1" rpy="0 0 0"/>
    </visual>
  </link>

  <link name="rotating_base">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.2"/>
      </geometry>
      <material name="blue"/>
      <origin xyz="0 0 -0.05" rpy="0 0 0"/>
    </visual>
  </link>

<joint name="rotation_joint" type="revolute">
    <parent link="base_link"/>
     <child link="rotating_base"/>
      <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
     <origin xyz="0 0 0.3" rpy="0 0 0"/>
    <axis xyz="0 0 1" />
  </joint>

  <link name="arm">
    <visual>
      <geometry>
        <cylinder length="${a2_length}" radius="0.05"/> <!-- Wczytaj z Parameter Server -->
      </geometry>
      <material name="yellow"/>
      <origin xyz="${a2_length/2} 0 0" rpy="0 1.57 0"/>
    </visual>
  </link>

<joint name="shoulder" type="revolute">
    <parent link="rotating_base"/>
     <child link="arm"/>
      <limit effort="1000.0" lower="0" upper="-1.57" velocity="0.5"/>
     <origin xyz="0 0 0" rpy="-1.57 0 0"/>
    <axis xyz="0 0 1" />
  </joint>

  <link name="forearm">
    <visual>
      <geometry>
        <cylinder length="${a3_length}" radius="0.05"/> <!-- Wczytaj z Parameter Server -->
      </geometry>
      <material name="green"/>
      <origin xyz="${a3_length/2} 0 0" rpy="0 1.57 0"/>
    </visual>
  </link>

<joint name="elbow" type="revolute">
    <parent link="arm"/>
     <child link="forearm"/>
      <limit effort="1000.0" lower="0.0" upper="1.57" velocity="0.5"/>
     <origin xyz="${a2_length} 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1" />
  </joint>

<link name="elo">
    <visual>
      <geometry>
        <cylinder length="0" radius="0.05"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0" rpy="0 1.57 0"/>
    </visual>
  </link>

<joint name="end" type="fixed">
    <parent link="forearm"/>
     <child link="elo"/>
      <limit effort="1000.0" lower="0.0" upper="1.57" velocity="0.5"/>
     <origin xyz="${a3_length} 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1" />
  </joint>

</robot>
